transform-processor/README.adoc

```diff
@@ -0,0 +1,36 @@
+= Spring Cloud Stream Transform Processor
+
+A *Spring Cloud Stream* module that allows A Processor module that allows transformation of messages
+using a SpEL expression or a Groovy script.
+
+== Requirements
+
+To run this sample, you will need to have installed:
+
+* Java 7 or Above
+
+This example requires Redis to be running on localhost.
+
+== Code Tour
+
+This project is a Spring Boot application that uses Spring Cloud Stream to
+bind module channels to external middleware.
+The Processor module has 2 primary components:
+
+* TransformProcessorApplication - the Spring Boot Main Application
+* TransformProcessor - the actual module code that interacts with message channels
+
+## Building with Maven
+
+Build the module by executing:
+
+```
+$> mvn package
+```
+
+## Running the Sample
```

Comment:
we need to stop using "sample" and "example" and "demo" in these readmes (and anywhere in the module code)

I actually simplified the READMEs for time-source and log-sink, and I think that's a decent format to use (with more detail as necessary):
https://github.com/spring-cloud/spring-cloud-stream-modules/tree/master/time-source
https://github.com/spring-cloud/spring-cloud-stream-modules/tree/master/log-sink